### PR TITLE
access top words in topic from S3. and new admin pages.

### PR DESCRIPTION
**Pull request by yingjieMiao**

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/learning/topicmodel/WordTopicStore.scala [hunk 1]
```diff
@@ -99,3 +105,29 @@ class S3WordStoreImpl(val bucketName: S3Bucket, val amazonS3Client: AmazonS3, va
...
+/**
+ * show top words related to topics
+ */
+trait TopicWordsStore extends ObjectStore[String, String]
+
+class S3TopicWordsStoreImpl(val bucketName: S3Bucket, val amazonS3Client: AmazonS3) extends S3ObjectStore[String, String] with TopicWordsStore {
+   val prefix = "word_topic/"
+
+  def unpackValue(s3Obj : S3Object) : String = {
+    val is = s3Obj.getObjectContent
```
Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/learning/topicmodel/WordTopicStore.scala:118]: style suggestions:

``` scala
val is = s3obj.getObjectContent
try {
  Source.fromInputStream(is, UTF8).getLines.mkString("\n")
} finally {
  is.close()
}
```

Also we usually write types like `s3obj: S3Object` (no space before the colon).

File: kifi-backend/modules/common/app/com/keepit/learning/topicmodel/WordTopicStore.scala [hunk 2]
```diff
@@ -99,3 +105,29 @@ class S3WordStoreImpl(val bucketName: S3Bucket, val amazonS3Client: AmazonS3, va
...
+      content
+    } finally {
+      is.close
+    }
+  }
+  def packValue(value : String) : (InputStream, ObjectMetadata) = {
+    throw new NotImplementedError       // currently this file is provided outside Scala. We are not supposed to write to the bucket.
+  }
+
+  def idToKey(id: String) = prefix + "%s.topicWords.txt".format(id)
```
Comment (gmethvin) [inline kifi-backend/modules/common/app/com/keepit/learning/topicmodel/WordTopicStore.scala:130]:
``` scala
s"${prefix}${id}.topicWords.txt"
```

File: kifi-backend/modules/shoebox/app/com/keepit/admin/TopicModelController.scala [hunk 3]
```diff
@@ -189,4 +190,40 @@ class TopicModelController  @Inject() (
...
+    }
+    log.info("getting topic words from S3")
+    val content = topicWordsStore.get(id).get
+    Ok(content)
+  }
+
+  // index is not necessarily the same as Id in DB. index starts from 1.
+  def viewTopicDetails(index: Int) = AdminHtmlAction{ implicit request =>
+    val topic = db.readOnly { implicit s =>
+      val topics = currentAccessor.topicNameRepo.all.sortWith((a, b) => a.id.get.id < b.id.get.id)
```
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/TopicModelController.scala:206]: You can also use `sortBy(_.id.get.id)` instead of `sortWith`.

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/TopicModelController.scala:206]: that's cool!

**Comment by gmethvin** [comment]
looks good to me

**Comment by yingjieMiao** [comment]
thanks for the comments!

**Pull request merged**